python/restate/serde.py

```diff
@@ -74,7 +74,24 @@ def _from_dict(data_class: typing.Any, data: typing.Any) -> typing.Any:  # pylin
         return _to_dict, _from_dict
 
 
+def try_import_msgspec_struct():
+    """
+    Try to import Struct from msgspec.
+    """
+    try:
+        from msgspec import Struct  # type: ignore # pylint: disable=import-outside-toplevel
+
+        return Struct
```

Comment:
Here you can also add a decode and encode functions and in the expect create their dummy variants.
with that you can avoid the inner import and it will be safe to import the `MsgspecJsonSerde`